File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/node/GenerateKeyResult.java
/*
 * © 2016-2019 FlowCrypt Limited. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.api.retrofit.response.node;

import android.os.Parcel;

import com.flowcrypt.email.api.retrofit.response.model.node.NodeKeyDetails;
import com.google.gson.annotations.Expose;

/**
 * It's a result for "generateKey" requests.
 *
 * @author Denis Bondarenko
 * Date: 4/1/19
 * Time: 9:44 AM
 * E-mail: DenBond7@gmail.com
 */
public class GenerateKeyResult extends BaseNodeResult {

  public static final Creator<GenerateKeyResult> CREATOR = new Creator<GenerateKeyResult>() {
    @Override
    public GenerateKeyResult createFromParcel(Parcel source) {
      return new GenerateKeyResult(source);
    }

    @Override
    public GenerateKeyResult[] newArray(int size) {
      return new GenerateKeyResult[size];
    }
  };

  @Expose
  private NodeKeyDetails key;

  public GenerateKeyResult() {
  }

  protected GenerateKeyResult(Parcel in) {
    super(in);
    this.key = in.readParcelable(NodeKeyDetails.class.getClassLoader());
  }

  @Override
  public int describeContents() {
    return 0;
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    super.writeToParcel(dest, flags);
    dest.writeParcelable(this.key, flags);
  }

  public NodeKeyDetails getKey() {
    return key;
  }
}

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/node/GenerateKeyResult.kt
/*
 * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.api.retrofit.response.node

import android.os.Parcel
import android.os.Parcelable
import com.flowcrypt.email.api.retrofit.response.base.ApiError
import com.flowcrypt.email.api.retrofit.response.model.node.NodeKeyDetails
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.BufferedInputStream

/**
 * It's a result for "generateKey" requests.
 *
 * @author Denis Bondarenko
 * Date: 4/1/19
 * Time: 9:44 AM
 * E-mail: DenBond7@gmail.com
 */
data class GenerateKeyResult constructor(@Expose val key: NodeKeyDetails?,
                                         @SerializedName("error")
                                         @Expose override val apiError: ApiError?) : BaseNodeResponse {
  override fun handleRawData(bufferedInputStream: BufferedInputStream) {

  }

  constructor(source: Parcel) : this(
      source.readParcelable<NodeKeyDetails>(NodeKeyDetails::class.java.classLoader),
      source.readParcelable<ApiError>(ApiError::class.java.classLoader)
  )

  override fun describeContents(): Int {
    return 0
  }

  override fun writeToParcel(dest: Parcel, flags: Int) =
      with(dest) {
        writeParcelable(key, flags)
        writeParcelable(apiError, flags)
      }

  companion object {
    @JvmField
    val CREATOR: Parcelable.Creator<GenerateKeyResult> = object : Parcelable.Creator<GenerateKeyResult> {
      override fun createFromParcel(source: Parcel): GenerateKeyResult = GenerateKeyResult(source)
      override fun newArray(size: Int): Array<GenerateKeyResult?> = arrayOfNulls(size)
    }
  }
}
